🚀 Debezium 1.0 release announcement (#398)
debezium/debezium.github.io: blog/2019-12-18-debezium-1-0-0-final-released.adoc

= Streaming Now: Debezium 1.0 Final Is Out!
gmorling
:awestruct-tags: [ releases, mysql, postgres, sqlserver, mongodb, cassandra, oracle ]
:awestruct-layout: blog-post

Today it's my great pleasure to announce the availability of Debezium *1.0.0.Final*!

Since the https://github.com/debezium/debezium/commit/0a99ed67cd8f74d6f451b0a2d3809e23127e4698[initial commit] in November 2015,
the Debezium community has worked tirelessly to realize the vision of building a comprehensive open-source low-latency platform for change data capture for a variety of databases.

> Within those four years, Debezium's feature set has tremendously grown: stable, highly configurable CDC connectors for MySQL, Postgres, MongoDB and SQL Server, incubating connectors for Apache Cassandra and Oracle, facilities for transforming and routing messages, support for design patterns such as the outbox pattern and much more.

Reviewer: "grown tremendously"

> Within those four years, Debezium's feature set has tremendously grown: stable, highly configurable CDC connectors for MySQL, Postgres, MongoDB and SQL Server, incubating connectors for Apache Cassandra and Oracle, facilities for transforming and routing messages, support for design patterns such as the outbox pattern and much more.

Author: Funny, it seems some of my mistakes sound so wrong to the native speaker's ear, that both you and @sjwoodman pointed them out :)

A very active, open and welcoming community of users, contributors and committers has formed around the project.
Debezium is deployed to production at lots of organizations from all kinds of industries,

> partly with huge installations, using hundreds of connectors to stream data changes out of thousands of databases.

Reviewer: some with huge installations

> partly with huge installations, using hundreds of connectors to stream data changes out of thousands of databases.

Author: Went for Simon's "some with".

The 1.0 release marks an important milestone for the project:
based on all the production feedback we got from the users of the 0.x versions, we figured it's about time to express the maturity of the four stable connectors in the version number, too.

== Why Debezium?

One of the things making it so enjoyable to work on Debezium as a tool for change data capture is the variety of potential use cases.
When presenting the project at conferences,

> it's just great to see how people quickly get excited when they realize all the possibilities enabled by Debezium and CDC.

Reviewer 2: Define CDC?

Author: Good point. I've changed the first occurence in the post to "change data capture (CDC)".

In a nutshell, Debezium is one big enabler for letting you react to changes in your data with a low latency.
Or, as one conference attendee recently put it, it's "like the observer pattern, but for your database".

Here's a few things we've seen Debezium being used for as a ingestion component in data streaming pipelines:

* Replicating data from production databases to other databases and data warehouses
* Feeding data to search services like Elasticsearch or Apache Solr
* Updating or invalidating caches

When using Debezium with Apache Kafka and its rich ecosystem of sink connectors
setting up such integrations can be done without any coding,
just by means of configuration.

> But there's many other use cases of CDC which go beyond just moving data from A to B.

Reviewer 2: ```suggestion
But there are many other use cases of CDC which go beyond just moving data from A to B.
```

When adding stream processing into the picture, e.g. via Kafka Streams or Apache Flink,
CDC enables you to run https://github.com/debezium/debezium-examples/tree/master/kstreams-live-update[time-windowed streaming queries], continuously updated as your data changes
("what's the aggregated order revenue per category within the last hour").
You can use CDC to link:/blog/2019/10/01/audit-logs-with-change-data-capture-and-stream-processing/[build audit logs] of your data,

> telling who did change which data items at which time.

Reviewer: telling who changed which data items at what time

> telling who did change which data items at which time.

Reviewer 2: ```suggestion
telling who changed which data items at what time.
```

> Or update denormalized views of your data, for the sake of efficient data retrieval, adhering to the CQRS pattern.

Reviewer 2: Should you define CQRS?

Author: Done.

> Finally, CDC also can play a vital role in microservices architectures;

Reviewer 2: ```suggestion
Finally, CDC can also play a vital role in microservices architectures;
```

exchanging data between services and keeping local views of data owned by other services achieves a higher independence, without having to rely on synchronous API calls.
One particularly interesting approach in this context is the link:/blog/2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern/[outbox pattern],

> which is link:/documentation/reference/1.0/configuration/outbox-event-router.html[supported well] by Debezium.

Reviewer 2: ```suggestion
which is link:/documentation/reference/1.0/configuration/outbox-event-router.html[well supported] by Debezium.
```

In case you don't start on the green field (who ever does?),

> CDC can be used to implement the strangler pattern for moving from a monolithic design to microservices.

Reviewer 2: link to something describing this pattern?

Author: 👍  Done.

You can learn more about change data capture use cases with Debezium and Apache Kafka in https://speakerdeck.com/gunnarmorling/practical-change-data-streaming-use-cases-with-apache-kafka-and-debezium-qcon-san-francisco-2019[this presentation] from QCon San Francisco.

But you don't have to take our word for it:
you can find lots of blog posts, conference talks and examples by folks using Debezium in production in our compilation of link:/documentation/online-resources/[resources].
If you'd like to get a glimpse of who else already is using Debezium,
see our rapidly growing link:/community/users/[list of reference users]
(or send us a https://github.com/debezium/debezium.github.io/blob/develop/community/users.asciidoc[pull request] to get your name added if your organization already is running Debezium in production).

== Debezium 1.0

Now, let's talk a little bit about the contents of the 1.0 release.

> This version continues the efforts we've begun in link:/blog/2019/10/02/debezium-0-10-0-final-released/[0.10] to make sure the emitted event structures and configuration options of the connectors are correct and consistent.

Reviewer 2: ```suggestion
This version continues the effort we began in link:/blog/2019/10/02/debezium-0-10-0-final-released/[0.10] to make sure the emitted event structures and configuration options of the connectors are correct and consistent.
```

While we've always been very careful to ensure a smooth upgrading experience, you can expect even more stability in this regard going forward after the 1.0 release.

We've expanded the test coverage of databases (Postgres 12, SQL Server 2019),

> build against the latest version of Apache Kafka (2.4.0) and upgraded our container images to OpenJDK 11.

Reviewer: build -> built

> build against the latest version of Apache Kafka (2.4.0) and upgraded our container images to OpenJDK 11.

Reviewer 2: The present tense of "build" conflicts with the past tense of "expanded" and "upgraded". But using "built" doesn't really work either. So how about 

We've expanded the test coverage of databases (Postgres 12, SQL Server 2019),
upgraded our container images to OpenJDK 11 and we build against the latest version of Apache Kafka (2.4.0).

which sounds better to me, even if not perfect.

Author: Thx, changed it along these lines.

You're amazing, and I would like to wholeheartedly thank each and everyone of you!

> I'm sure our community is continuing to grow in the future -- I'd love if we hit the mark of 200 contributors in 2020.

Reviewer 2: ```suggestion
I'm sure our community will continue to grow in the future -- I'd love it if we hit the mark of 200 contributors in 2020.
```

Equally important are our users; interacting with you in the chat, on the mailing list or at conferences and meet-ups is what helps to drive the direction of the project:
learning about your specific requirements and use cases (or bugs you've run into) is vital for deciding where to put the focus next.
A big thank you to you, too!

> Some of you even have shared their experiences with Debezium in conference talks and blog posts.

Reviewer 2: ```suggestion
Some of you have even shared their experiences with Debezium in conference talks and blog posts.
```

> Nothing beats to hear the war stories from others and being able to learn from their experiences,

Reviewer: "Nothing beats hearing the war stories shared by others"

> Nothing beats to hear the war stories from others and being able to learn from their experiences,

Reviewer 2: ```suggestion
Nothing beats hearing the war stories of others and being able to learn from their experiences,
```

so you speaking about your insights around Debezium and CDC is incredibly helpful and highly appreciated!

== What's Next?

> Let's wrap up this post with an outlook to what's next in store for Debezium.

Reviewer 2: ```suggestion
Let's wrap up this post with a look to see what's next in store for Debezium.
```

Of course, this roadmap is strongly influenced by the community, i.e. you.
So if you would like to see any particular items here, please let us know.

We also have some exciting blog posts in the workings,
e.g. on how to combine Debezium with the brand-new Kafka Connect https://camel.apache.org/blog/Camel-Kafka-connector-intro/[connector for Apache Camel] or how to use the recently added support for non-key joins in Kafka Streams (https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=74684836[KIP-213]) with Debezium change events.

One more thing I'm super-thrilled about is Debezium becoming a supported component of the https://www.redhat.com/en/blog/whats-new-red-hat-integration[Red Hat Integration] product.
Part of the current release is a https://developers.redhat.com/blog/2019/11/22/red-hat-advances-debezium-cdc-connectors-for-apache-kafka-support-to-technical-preview/[Tech Preview] for the change data capture connectors for MySQL, Postgres, SQL Server and MongoDB.
This is great news for folks who wish to have commercial support by Red Hat for their CDC connectors.

For now, let's celebrate the release of Debezium 1.0 and look forward to what's coming in 2020.

> Upwards and Onwards!

Reviewer 2: It's usually the other way around:

```suggestion
Onwards and Upwards!
```
